test/DynamoCoreTests/libGPreloaderTests.cs

```diff
@@ -204,6 +204,104 @@ public void GetLibGPreloaderLocation_libGVersionFallback()
+            //assert the non tolerant version throws
+            Assert.Throws<DirectoryNotFoundException>(() => { DynamoShapeManager.Utilities.GetGeometryFactoryPath2(rootFolder, targetVersion); });
+
+            // cleanup
+            libG22440path.Delete(true);
+            libG22500path.Delete(true);
+        }
+
+        [Test]
+        public void GetGeometryFactoryPathTolerant_NoMatch()
+        {
+            var versions = new List<Version>()
+            {
+                    new Version(225,0,0)
+            };
+
+            var mockedInstalledASMs = new Dictionary<string, Tuple<int, int, int, int>>()
+            {
+
+                {"someInstallWithNoMatchingASM" ,Tuple.Create<int,int,int,int>(224,0,0,0)},
+            };
+
+            var targetVersion = new Version(225, 2, 0);
+
+            // mock a folder with libASMLibVersionToVersion folders with correct names
+            var foundPath = "";
+            var rootFolder = Path.Combine(Path.GetTempPath(), "LibGTest");
+            // both versions of libG exist
+            var libG22500path = System.IO.Directory.CreateDirectory(Path.Combine(rootFolder, "LibG_225_0_0"));
+            var libGTestPath = new DirectoryInfo(Path.Combine(rootFolder, "LibG_224_24_24"));
+            var foundVersion = DynamoShapeManager.Utilities.GetInstalledAsmVersion2(
+                versions, ref foundPath, rootFolder, (path) => { return mockedInstalledASMs; });
+
+            // There is no match, so found version is null.
+            Assert.AreEqual(null, foundVersion);
+            // There is no match, so path to ASM is null.
+            Assert.AreEqual(string.Empty, foundPath);
+
+            // when passed null as a version, GetLibGPreloaderLocation will return LibG_0_0_0
+            Assert.IsTrue(DynamoShapeManager.Utilities.GetLibGPreloaderLocation(foundVersion, rootFolder).ToLower().Contains("libg_0_0_0"));
+
+
+            //when passed a null version this method should return empty string - as the geometry path was not located
+            Assert.AreEqual(string.Empty,
+                DynamoShapeManager.Utilities.GetGeometryFactoryPathTolerant(rootFolder, null));
+
+            //when passed a null root directory this method should return empty string - as a valid root directory is required.
+            Assert.AreEqual(string.Empty,
+                DynamoShapeManager.Utilities.GetGeometryFactoryPathTolerant(null, new Version(224, 24, 24)));
+
+            // when passed a non null, non matching version this method should return a path to the version that was supplied - 
+            // but it will likely fail later.
+            Assert.AreEqual(Path.Combine(libGTestPath.FullName, DynamoShapeManager.Utilities.GeometryFactoryAssembly).ToLower(),
+                 DynamoShapeManager.Utilities.GetGeometryFactoryPathTolerant(rootFolder, new Version(224,24,24)).ToLower());
```

Comment:
Why not make this throw an exception just like `GetGeometryFactoryPath2` instead? It looks like returning a non-existent location from here will just delay the user in recognizing an error.